### ISSUE TITLE
DisallowVariableParsing: disallowSimpleSyntax false positive for `"{$a[$b]}"`

### ISSUE THREAD

**Issue by LastDragon-ru** [slevomat/coding-standard#1553]
```xml
<?xml version="1.0"?>
<ruleset name="LaraAspStandard"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="./vendor/squizlabs/php_codesniffer/phpcs.xsd">
    <rule ref="SlevomatCodingStandard.Strings.DisallowVariableParsing">
        <properties>
            <property name="disallowSimpleSyntax" value="true"/>
            <property name="disallowDollarCurlySyntax" value="true"/>
            <property name="disallowCurlyDollarSyntax" value="false"/>
        </properties>
    </rule>
</ruleset>
```
```php
<?php declare(strict_types = 1);

namespace tmp;

class Bug {
    public static function bug(): void {
        $a = [0 => 'a'];
        $b = 0;

        echo "$b";       // error    -> ok
        echo "{$b}";     // no error -> ok
        echo "{$a[$b]}"; // error    -> false positive
    }
}
```
```txt
------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
------------------------------------------------------------------------------------
 10 | ERROR | Using variable syntax "$..." inside string is disallowed, found "$b".
 12 | ERROR | Using variable syntax "$..." inside string is disallowed, found "$b".
------------------------------------------------------------------------------------
```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/3cddc977038419e29381cfa2c6c56201951e6806
